src/main/java/org/broadinstitute/hellbender/utils/tsv/SimpleCSVWriterWrapperWithHeader.java

```diff
@@ -0,0 +1,197 @@
+     * rows of the CSV.
+     *
+     * @param columns Ordered list of header lines to be built into the CSV
+     */
+    public void setHeaderLine(List<String> columns) {
+        if (headerMap != null) {
+            throw new GATKException("Should not be adding multiple header lines to a file");
+        }
+        outputWriter.writeNext(columns.toArray(new String[0]), false);
+        expectedColumns = columns.size();
+
+        // Create the mapping between header and column
+        headerMap = new HashMap<>();
+        for (int i = 0; i < columns.size(); i++) {
+            Utils.nonNull(columns.get(i));
+            if (headerMap.putIfAbsent(columns.get(i), i) != null) {
+                throw new GATKException("Only allow unique column headings");
+            }
+        }
+    }
+
+    private void writeLine(String[] line) {
+        outputWriter.writeNext(line, false);
+        currentLine = null;
+    }
+
+    /**
+     * Builds a new SimpleCSVWriterLineBuilder and writes out the previous line if it exists.
+     *
+     * @return a blank SimpleCSVWriterLineBuilder to allow for defining the next line
+     */
+    public SimpleCSVWriterLineBuilder getNewLineBuilder() {
+        if (headerMap == null) {
+            throw new GATKException("Cannot construct line without first setting the header line");
+        }
+        if (currentLine != null) {
+            currentLine.write();
+        }
+        currentLine = new SimpleCSVWriterLineBuilder(this, expectedColumns);
+        return currentLine;
+    }
+
+    /**
+     * @param column header line to get index for
+     * @return zero based index corresponding to that header string, throws an exception if the headerline doesn't exist
+     */
+    public Integer getIndexForColumn(String column) {
+        Utils.nonNull(headerMap, "Cannot request column index if the header has not been specified");
+        Integer index = headerMap.get(column);
+        Utils.nonNull(index, "Requested column " + column + " does not exist in the provided header");
+        return index;
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (currentLine != null) {
+            currentLine.write();
+        }
+        outputWriter.close();
+    }
+
+    /**
+     * Helper to allow for incremental construction of a body line using either indexes or column headings
+     * <p>
+     * Calling build() will cause the line to be written out into the underlying CSV writer in its current state. Doing
+     * so will result in a validation call where an exception will be thrown if any columns of the current line have
+     * not been defined. fill() can be used to provide a default value for undefined columns.
+     */
+    public class SimpleCSVWriterLineBuilder {
+        SimpleCSVWriterWrapperWithHeader thisBuilder;
+        String[] lineToBuild;
+        boolean hasBuilt = false;
+
+        SimpleCSVWriterLineBuilder(SimpleCSVWriterWrapperWithHeader me, int lineLength) {
```

Comment:
As a non-static inner class, you can just directly access the fields of the enclosing class.

Reply:
`lineLength` -> `numberOfColumns`